test/recordlentest.c

```diff
@@ -74,6 +74,13 @@ static int fail_due_to_record_overflow(int enc)
             && ERR_GET_REASON(err) == reason)
         return 1;
 
+#if SSL3_ALIGN_PAYLOAD < 2
+    if (ERR_GET_LIB(err) == ERR_LIB_SSL
+            && ERR_GET_REASON(err) == SSL_R_PACKET_LENGTH_TOO_LONG
+            && reason == SSL_R_ENCRYPTED_LENGTH_TOO_LONG)
+        return 1;
+#endif
```

Comment:
I think a comment here explaining why this block is necessary would be worthwhile.

Reply:
Actually, I think the need for this might be indicating another bug....investigating...

Reply:
Indeed...the test in `tls_get_more_records` which returns `SSL_R_PACKET_LENGTH_TOO_LONG` is failing to properly account for the `rbuf->offset` value. If it did so, we would be seeing the packet length too long error, even for other values of `SSL3_ALIGN_PAYLOAD`